pkg/customsignaturestore/customsignaturestore.go

```diff
@@ -104,14 +123,106 @@ func (s *Store) String() string {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	if s.customURIs == nil {
-		return "ClusterVersion signatureStores unset, falling back to default stores"
-	} else if len(s.customURIs) == 0 {
+	if s.customStores == nil {
+		return "ClusterVersion signatureStores not set, falling back to default stores"
+	} else if len(s.customStores) == 0 {
 		return "0 ClusterVersion signatureStores"
 	}
-	uris := make([]string, 0, len(s.customURIs))
-	for _, uri := range s.customURIs {
-		uris = append(uris, uri.String())
+	customStores := make([]string, 0, len(s.customStores))
+	for _, customStore := range s.customStores {
+		caName := customStore.CA.Name
+		if caName != "" {
+			customStores = append(customStores, caName)
+		}
+	}
+	return fmt.Sprintf("ClusterVersion signatureStores: %s", strings.Join(customStores, ", "))
+}
+
+// HTTPClient provides a method for generating an HTTP client
+// with the proxy and trust settings, if set in the cluster.
+func (s *Store) HTTPClient(certAuthName string) (*http.Client, error) {
+	transportOption, err := s.getTransport(certAuthName)
+	if err != nil {
+		return nil, err
+	}
+	transportConfig := &transport.Config{Transport: transportOption}
+	transport, err := transport.New(transportConfig)
+	if err != nil {
+		return nil, err
+	}
+	return &http.Client{
+		Transport: transport,
+	}, nil
+}
+
+// getTransport constructs an HTTP transport configuration, including
+// any custom proxy configuration.
+func (s *Store) getTransport(certAuthName string) (*http.Transport, error) {
+	transport := &http.Transport{}
+
+	proxyConfig, err := s.getProxyConfig()
+	if err != nil {
+		return transport, err
+	} else if proxyConfig != nil {
```

Comment:
these two `if`s are unrelated so lets keep them separate?

```suggestion
	} 
	if proxyConfig != nil {
```

Reply:
Done in 006fd24d4332 -> 4ead92af5e0927; I'm marking this thread resolved.